include/particle.tcc

```diff
@@ -719,3 +724,20 @@ void mpm::Particle<Tdim, Tnphases>::apply_particle_velocity_constraints() {
     this->velocity_(direction, phase) = constraint.second;
   }
 }
+
+//! Apply particle velocity constraints
+template <unsigned Tdim, unsigned Tnphases>
+Eigen::VectorXd mpm::Particle<Tdim, Tnphases>::vector_data(
+    unsigned phase, const std::string& property) {
+  //! Map of properties
+  std::map<std::string, std::function<Eigen::VectorXd(unsigned)>> properties;
```

Comment:
I wonder whether we should make `map` every time we call `vector_data`. My thoughts would be to have the map as private variables.

Reply:
the problem is we can't pass phase otherwise?

Reply:
I see. How about a map that has two keys (attribute and phase), like a tuple? We can define the phase up to 3 only right, not infinite.

I still feel making map for every particle and every time step we output is a bit inefficient.